src/cmap/handshake/client_metadata.ts

```diff
@@ -234,3 +234,58 @@ export function getFAASEnv(): Map<string, string | Int32> | null {
 
   return null;
 }
+
+/**
+ * @internal
+ * This type represents the global Deno object and the minimal type contract we
+ * expect it to satisfy. In order to not ship code in the driver that would break
+ * future versions of this runtime assume all properties are nullish.
+ */
+declare const Deno: { version?: { deno?: string } } | undefined;
+
+/**
+ * @internal
+ * This type represents the global Bun object and the minimal type contract we
+ * expect it to satisfy. In order to not ship code in the driver that would break
+ * future versions of this runtime assume all properties are nullish.
+ */
+declare const Bun: { (): void; version?: string } | undefined;
+
+/**
+ * @internal
+ * Get current JavaScript runtime platform
+ *
+ * NOTE: The version information fetching is intentionally written aggressively defensive
+ * to avoid having a released driver version that becomes incompatible
+ * with a future change to these global objects
+ */
+function getRuntimeInfo(): string {
+  if ('Deno' in globalThis) {
```

Comment:
Why are all the checks necessary? Can't we assume if `Deno` or `Bun` are present in the global namespace that they conform to the type we expect? And that if a user was to randomly put them there all bets are off?

Reply:
We just want to write code that won't end up throwing if there's a potential change. We can reduce the checks here to optional chains though